packages/core/src/exchanges/ssr.ts

```diff
@@ -159,7 +159,16 @@ export const ssrExchange = (params?: SSRExchangeParams): SSRExchange => {
     return merge([forwardedOps$, cachedOps$]);
   };
 
-  ssr.restoreData = (restore: SSRData) => Object.assign(data, restore);
+  ssr.restoreData = (restore: SSRData) => {
+    // Only restore data for uncached operations
+    const uncached = {};
+    Object.entries(restore).forEach(([key, value]) => {
+      if (false === key in data) {
+        uncached[key] = value;
+      }
+    });
+    return Object.assign(data, uncached);
+  };
```

Comment:
Hmm, this kinda feels weird to me. Generally the same key in the cache should have the same result across different pages, I'm curious how and why this would deviate.

This would be a breaking change if we were to go for this.

Reply:
Maybe it would make more sense to put this in
https://github.com/FormidableLabs/urql/blob/891a17a0a22e3b4d5b301a779330336efc79d1b0/packages/next-urql/src/with-urql-client.ts#L58-L60
so it only affects the next integration?